File: 6-kyu-kata/multiples-of-3-or-5/typescript/multiplesOf3Or5.Test.ts
// 6 kyu - Multiples of 3 or 5  [ ID: 514b92a657cdc65150000006  (multiples-of-3-or-5) ]
// URL: https://www.codewars.com/kata/514b92a657cdc65150000006
// Category: ALGORITHMS  |  Tags: MATHEMATICS | ALGORITHMS
// *****************************************************************************
import solution = require("./solution")
import { assert } from "chai"

function test(num: number, expected: number) {
  assert.strictEqual(solution.Challenge.solution(num), expected)
}

function _solution(number: number) {
  var sum = 0
  for (var i = 0; i < number; i++) {
    if (i % 3 == 0 || i % 5 == 0) {
      sum = sum + i
    }
  }
  return sum
}

describe("solution", function () {
  it("should handle basic tests", function () {
    test(10, 23)
    test(20, 78)
    test(200, 9168)
  })
  it("should handle the smallest cases", function () {
    test(-1, 0)
    test(0, 0)
    test(1, 0)
    test(2, 0)
    test(3, 0)
    test(4, 3)
    test(5, 3)
    test(6, 8)
  })
  it("should handle random cases", function () {
    for (var i = 0; i < 10; i++) {
      let ran = Math.floor(Math.random() * 200)
      test(ran, _solution(ran))
    }
  })
})

File: 6-kyu-kata/multiples-of-3-or-5/swift/multiplesOf3Or5.Test.swift
// 6 kyu - Multiples of 3 or 5  [ ID: 514b92a657cdc65150000006  (multiples-of-3-or-5) ]
// URL: https://www.codewars.com/kata/514b92a657cdc65150000006
// Category: ALGORITHMS  |  Tags: MATHEMATICS | ALGORITHMS
// *****************************************************************************
import XCTest
import Glibc

class SolutionTest: XCTestCase {
    static var allTests = [
        ("Test From Example", testExampleSolution),
        ("Test Zero", testZero),
        ("Test Large Solutions", testLargeSolutions),
    ]

    func testExampleSolution() {
        XCTAssertEqual(solution(10), 23)
        XCTAssertEqual(solution(20), 78)
        XCTAssertEqual(solution(200), 9168)
        XCTAssertEqual(solution(300), 20850)
        XCTAssertEqual(solution(320), 23628)
    }

    func testZero() {
        XCTAssertEqual(solution(0), 0)
    }

    func testLargeSolutions() {
        XCTAssertEqual(solution(10865), 27535533)
    }
}

XCTMain([
    testCase(SolutionTest.allTests)
])
